### ISSUE TITLE
Support inserting multiple documents using current database time

### ISSUE THREAD

**Issue by dpacosta** [awslabs/amazon-qldb-driver-nodejs#97]
### Library version being used
2.2.0

---

### Use case
Insert multiple documents and indicate when they were created using the database time. 

---

### Experiments

1.  [Using a list of documents](https://docs.aws.amazon.com/qldb/latest/developerguide/driver-cookbook-nodejs.html#cookbook-nodejs.crud.inserting.multiple)

`txn.execute("INSERT INTO People ?", [ { field1: 'value1', field2: 'value2', createdAt: 'UTCNOW()' },  { field1: 'value1', field2: 'value2', createdAt: 'UTCNOW()' }]);`

**Problem**: `UTCNOW()` is saved as a string

2.  Building the raw insert command and passing it in to the `execute` method along with its field-level parameters

`txn.execute("INSERT INTO People << { 'field1': ?, 'field2': ?, 'createdAt': ? },  { 'field1': ?, 'field2': ?, 'createdAt': ? } >>', 'value1', 'value2', 'UTCNOW()', 'value3', 'value4', 'UTCNOW()'");`

**Problem**: The documentation does not mention this can be done but I tried anyway since it seemed quite intuitive. I got `BadRequestException: Query expects X parameters, but statement has Y` type of error so it confirmed that it is indeed not a fit for this use case.

---

### Undesirable way out
Write my own raw query builder for inserting multiple documents, which introduces injection risks.

---

### Ask
Add the ability to insert multiple documents using the database time to set values in a safe way.

**Comment by PlasmaIntec** [comment]
Hi @dpacosta. Can you clarify whether your use case is to access the timestamp when a certain document was created?

For example, the following query retrieves the timestamp when a document revision is committed:
```
SELECT h.metadata.txTime FROM history(People) AS h
WHERE h.data.id = 'ADR2Ll1fGsU4Jr4EqTdnQF'
```
To find the creation timestamp, you can sort result list and take the earliest timestamp.

Alternatively, you can remove the single quotes around `UTCNOW()` and hardcode it in the query, like in the following:
```
txn.execute("INSERT INTO People << { 'firstName': ?, 'createdAt': UTCNOW() }, { 'firstName': ?, 'createdAt': UTCNOW() } >>", "Joe", "Jane")
```

**Comment by dpacosta** [comment]
@PlasmaIntec I tried what you suggested before and it was throwing `BadRequestException: Query expects X parameters, but statement has Y` like I mentioned on the original post. I jumped to the conclusion that the library was lacking the ability to handle the combination of multiple documents + multiple arguments , but I just double checked and turns out it was a misuse of the `execute` method, I was passing the arguments wrapped in a list instead and that was the problem, the method is supposed to get a variable number of arguments and not a list. I fixed it and now things are working.

I'm closing this issue.
